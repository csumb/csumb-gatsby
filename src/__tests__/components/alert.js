import React from 'react'
import renderer from 'react-test-renderer'
import {
  AlertDanger,
  AlertSuccess,
  AlertWarning,
  AlertEmpty,
  AlertFyi,
} from '../../components/alert'

describe('Alert', () => {
  it('renders a danger alert correctly', () => {
    const tree = renderer
      .create(
        <AlertDanger>
          <p>This is an alert</p>
        </AlertDanger>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a warning alert correctly', () => {
    const tree = renderer
      .create(
        <AlertWarning>
          <p>This is an alert</p>
        </AlertWarning>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an info alert correctly', () => {
    const tree = renderer
      .create(
        <AlertSuccess>
          <p>This is an alert</p>
        </AlertSuccess>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an empty alert correctly', () => {
    const tree = renderer
      .create(
        <AlertEmpty>
          <p>This is an alert</p>
        </AlertEmpty>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an fyi alert correctly', () => {
    const tree = renderer
      .create(
        <AlertFyi>
          <p>This is an alert</p>
        </AlertFyi>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
