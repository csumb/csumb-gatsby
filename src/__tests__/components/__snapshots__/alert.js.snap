// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders a danger alert correctly 1`] = `
<div
  className="css-xtr3pa"
  data-reach-alert={true}
>
  <div
    className="css-1p349jc"
  >
    Danger
  </div>
  <div
    className="sc-bdVaJa sc-bwzfXH hbZeap"
  >
    <div
      className="css-n80ioy sc-bdVaJa iqCHRR"
      width={
        Array [
          0,
          0.16666666666666666,
          0.1,
        ]
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 css-1ule52c"
        data-icon="exclamation-circle"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="sc-bdVaJa kkdbJs"
      width={
        Array [
          1,
          0.8333333333333334,
          0.9,
        ]
      }
    >
      <p>
        This is an alert
      </p>
    </div>
  </div>
</div>
`;

exports[`Alert renders a warning alert correctly 1`] = `
<div
  className="css-jcln01"
  data-reach-alert={true}
>
  <div
    className="css-1p349jc"
  >
    Warning
  </div>
  <div
    className="sc-bdVaJa sc-bwzfXH hbZeap"
  >
    <div
      className="css-n80ioy sc-bdVaJa iqCHRR"
      width={
        Array [
          0,
          0.16666666666666666,
          0.1,
        ]
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-triangle fa-w-18 css-1ule52c"
        data-icon="exclamation-triangle"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="sc-bdVaJa kkdbJs"
      width={
        Array [
          1,
          0.8333333333333334,
          0.9,
        ]
      }
    >
      <p>
        This is an alert
      </p>
    </div>
  </div>
</div>
`;

exports[`Alert renders an empty alert correctly 1`] = `
<div
  className="css-19b7sg5"
  data-reach-alert={true}
>
  <div
    className="css-1p349jc"
  >
    For your information
  </div>
  <div
    className="sc-bdVaJa sc-bwzfXH hbZeap"
  >
    <div
      className="css-n80ioy sc-bdVaJa iqCHRR"
      width={
        Array [
          0,
          0.16666666666666666,
          0.1,
        ]
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-times-circle fa-w-16 css-1ule52c"
        data-icon="times-circle"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="sc-bdVaJa kkdbJs"
      width={
        Array [
          1,
          0.8333333333333334,
          0.9,
        ]
      }
    >
      <p>
        This is an alert
      </p>
    </div>
  </div>
</div>
`;

exports[`Alert renders an fyi alert correctly 1`] = `
<div
  className="css-2173aj"
  data-reach-alert={true}
>
  <p>
    This is an alert
  </p>
</div>
`;

exports[`Alert renders an info alert correctly 1`] = `
<div
  className="css-gnh87"
  data-reach-alert={true}
>
  <div
    className="css-1p349jc"
  >
    For your information
  </div>
  <div
    className="sc-bdVaJa sc-bwzfXH hbZeap"
  >
    <div
      className="css-n80ioy sc-bdVaJa iqCHRR"
      width={
        Array [
          0,
          0.16666666666666666,
          0.1,
        ]
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check-circle fa-w-16 css-1ule52c"
        data-icon="check-circle"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="sc-bdVaJa kkdbJs"
      width={
        Array [
          1,
          0.8333333333333334,
          0.9,
        ]
      }
    >
      <p>
        This is an alert
      </p>
    </div>
  </div>
</div>
`;
